src/main/java/org/truffleruby/language/exceptions/TopLevelRaiseHandler.java

```diff
@@ -80,4 +88,13 @@ public void setLastException(VirtualFrame frame, DynamicObject exception) {
         setExceptionVariableNode.setLastException(frame, exception);
     }
 
+    public void handleSignalException(DynamicObject exception) {
+        if (Layouts.BASIC_OBJECT.getLogicalClass(exception) == coreLibrary().getSignalExceptionClass()) {
+            getContext().getSafepointManager().pauseAllThreadsAndExecute(this, false, (thread, current_node) -> {
+                CallDispatchHeadNode node = CallDispatchHeadNode.createPrivate();
```

Comment:
Should be `CallDispatchHeadNode.getUncached()`

Reply:
Fixed.